### DISCUSSION TITLE
Is it possible to do precheck on my local metadata?

### DISCUSSION THREAD

**Discussion by mflint** [fastlane/fastlane#19048] [Q&A] [answered]
# tl;dr

I'd like to run `precheck` on my _local_ metadata, not the metadata in AppStoreConnect.

Is there any way to do this?

# Why?

I have a little CI pipeline with stages "test", "upload to testflight", "screenshots", "submit for review". So `precheck` is the very last thing to run - so if my metadata is bad, then it takes a long time to fail.

Ideally I'd be able to run `precheck` in my "test" stage, so I'll know quickly if I need to fix my metadata.

**Comment by crazymanish** [comment, marked as answer]
Hi, Fastlane `Precheck` is always check the AppStoreConnect version meta-data. 

I will suggest you to tweak your CI pipeline a little bit as below:
- test
- upload_metadata with `precheck` option (using deliver without screenshot and binary)

once above are good ✅ Do below
- upload_to_testflight
- upload_screenshots
- submit_for_review (without precheck, because we already done it earlier)

--------------
btw, we can run `precheck` local using custom action or plugin, because all the `precheck` rules are accessible/public, you can do that too...but might be more work 😉
